plugins/module_utils/network/sonic/config/snmp/snmp.py

```diff
@@ -0,0 +1,1400 @@
+
+        payload['ietf-snmp-ext:notifications'] = enable_trap_dict
+        trap_enable_request = {'path': enable_trap_url, 'method': method, 'data': payload}
+        requests.append(trap_enable_request)
+        return requests
+
+    def build_create_group_payload(self, config):
+        """ Build the payload for SNMP group
+        :rtype: A dictionary
+        :returns: The payload for SNMP group
+        """
+        payload_url = {}
+        group_list = []
+        group = config.get('group')
+        group_payload = {}
+        for conf in group:
+            group_dict = {}
+            if conf.get('name') is None:
+                break
+            group_dict['name'] = conf.get('name')
+            group_dict['access'] = self.build_create_group_access_payload(conf)
+            group_list.append(group_dict)
+        group_payload['group'] = group_list
+        payload_url['ietf-snmp:vacm'] = group_payload
+        return payload_url
+
+    def build_create_group_access_payload(self, config):
+        """ Build the payload for the SNMP group access suboption
+        :rtype: A list of dictionaries
+        :returns: the list of access
+        """
+        access_list = []
+        access_dicts = config.get('access')
+        if access_dicts is None:
+            return access_list
+        for access in access_dicts:
+            access_dict = {}
+            access_dict['context'] = 'Default'
+            access_dict['notify-view'] = access.get('notify_view')
+            access_dict['read-view'] = access.get('read_view')
+            access_dict['write-view'] = access.get('write_view')
+            security_level = access.get('security_level')
+            security_model = access.get('security_model')
+            if security_level:
+                access_dict['security-level'] = security_level
+            access_dict['security-model'] = security_model
+            access_list.append(access_dict)
+
+        return access_list
+
+    def build_create_enable_target_payload(self, config, have, overridden_or_replaced):
+        """ Build the payload for SNMP target information based on the given host configuration
+        :rtype: A dictionary
+        :returns: The payload for SNMP target
+        """
+        payload_url = {}
+        target_list = []
+        target = config.get('host')
+        have_targetentry = have.get('host')
+        have_targetentry_names = []
+        if have_targetentry and not overridden_or_replaced:
+            have_targetentry_names = [have_entry.get('name') for have_entry in have_targetentry]
+
+        for conf in target:
+            target_dict = {}
+            retries = conf.get('retries')
+            if retries is None:
+                retries = 3
+            target_entry_name = ""
+            if 'name' in conf:
+                target_entry_name = conf.get('name')
+            else:
+                target_entry_name = self.get_targetentry(have_targetentry_names)
```

Comment:
The name will have been found or created by build_create_enable_target_params_payload and saved in the config, so avoid allocating another one here.

```suggestion
            target_entry_name = conf['name']
```

Reply:
I am leaving this "Issue" in an unresolved state for now to help in tracking this request for changes. (The current proposed alternative to this suggestion won't work as intended.)